File: week2/problemSet2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string words = get_string("Please enter the text you want to analyze: ");
    float letters = count_letters(words);
    float num_words = count_words(words);
    float sentences = count_sentences(words);
    float L = (letters/num_words) * 100;
    float S = (sentences/num_words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    // printf("%f\n", letters);
    // printf("%f\n", num_words);
    // printf("%f\n", sentences);
    // printf("%f\n", L);
    // printf("%f\n", S);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int character, count = 0;
    //I want to count all letters (only), no punctuation or spaces
    for (int i = 0; i < strlen(text); i++)
    {
        character = (int) tolower(text[i]) - 97;
        if(character <= 25 && character >= 0)
        {
            count++;
        }
    }
    return count;
}

//This will return the number of words
int count_words(string text)
{
    int character, count = 0;
    //Counting only spaces
    for (int i = 0; i < strlen(text); i++)
    {
        character = (int) text[i];
        if (character == 32)
        {
            count++;
        }
    }
    return count + 1;
}

int count_sentences(string text)
{
    int character, count = 0;
    //Counting only periods, exclamation marks or questions
    for (int i = 0; i < strlen(text); i++)
    {
        character = (int) text[i];
        if (character == 46 || character == 33 || character == 63)
        {
            count++;
        }
    }
    return count;
}

//Passed 11/11 tests and style score of 96%

File: week6/PSet6/readability.py
# TODO

#Get text input from the user
text = input("Text: ")
text = text.lower()
#Initialize the variables
words = 1
sentences = 0
letters = 0

for letter in text:
    if(letter == ' '):
        words += 1
    elif letter == '?' or letter == '.' or letter == '!':
        sentences += 1
    #counting all non alpha characters
    elif ord(letter) < 96 or ord(letter) > 123:
        letters += 1
    else:
        continue

alpha_char = len(text) - words - sentences - letters + 1
el = (alpha_char / words) * 100
es = (sentences / words) * 100

index = (0.0588 * el) - (0.296 * es) - 15.8
if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {int(round(index, 0))}")


#passed 10/10 tests and 88% on the styling